R/assign_missingness.R

```diff
@@ -43,28 +47,50 @@
 #'   retain_columns = c(pg_protein_accessions)
 #' )
 #' }
-assign_missingness <- function(data, sample, condition, grouping, intensity, ref_condition = "control", completeness_MAR = 0.7, completeness_MNAR = 0.20, retain_columns = NULL) {
+assign_missingness <- function(data, sample, condition, grouping, intensity, ref_condition = "all", completeness_MAR = 0.7, completeness_MNAR = 0.20, retain_columns = NULL) {
   . <- NULL
-  conditions_no_ref <- unique(pull(data, !!ensym(condition)))[!unique(pull(data, !!ensym(condition))) %in% ref_condition]
-  if (!(ref_condition %in% unique(pull(data, {{ condition }})))) {
+  if (!(ref_condition %in% unique(dplyr::pull(data, {{ condition }}))) & ref_condition != "all") {
     stop("The name provided to ref_condition cannot be found in your conditions! Please provide a valid reference condition.")
   }
+  
+  if(ref_condition == "all"){
+    # creating all pairwise comparisons
+    all_conditions <- unique(dplyr::pull(data, {{ condition }}))
+    
+    all_combinations <- tibble::as_tibble(t(utils::combn(all_conditions, m = 2))) %>% 
+      dplyr::mutate(combinations = paste0(.data$V1, "_vs_", .data$V2)) 
+    
+    message('"all" was provided as reference condition. All pairwise comparisons are created from the conditions and assigned their missingness.\n The created comparisons are: \n', paste(all_combinations$combinations, collapse =  "\n"))
```

Comment:
Maybe change this to "pairwise comparisons are created from all conditions and their missigness type is assigned."